code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -145,6 +145,9 @@ Contains most of the procs that are called when a mob is attacked by something
 
 //Returns 1 if the attack hit, 0 if it missed.
 /mob/living/carbon/human/attacked_by(obj/item/I, mob/living/user, def_zone)
+	if(status_flags & INCORPOREAL) //We can't physically attack the incorporeal
+		return FALSE
+
```

Comment:
You shouldnt check this twice, either th object should check before attacking, or this code should happen